File: Hour 2/Q2/FirstNonRepeat.py
st = 'srtrwffsrrt'


def FirstNonRepeat(st):
    arr = list(st)
    freq = [0]*255
    for i in range(len(arr)):
        freq[ord(arr[i])] += 1
    for item in arr:
        if freq[ord(item)] == 1:
            return item
    return '0'


print(FirstNonRepeat(st))

File: Hour 2/Q2/FirstNonRepeat.cpp
#include<iostream>
using namespace std;

char FirstNonRepeat(string str){
    int i, freq[256] = {0};
    for(i = 0; i < str.length(); i++)
        freq[int(str[i])]++;
    for(i = 0; i < str.length(); i++){
        if(freq[int(str[i])] == 1)
            return str[i];
    }
    return '0';
}

int main(){
    string str = "nojon";
    cout<<FirstNonRepeat(str);
    return 0;
}